src/pykeen/nn/representation.py

```diff
@@ -947,13 +949,21 @@ def __init__(
     @classmethod
     def from_triples_factory(
         cls,
-        triples_factory: TriplesFactory,
+        *,
+        dataset: Optional[Dataset] = None,
```

Comment:
I'd rather have a separate constructor for dataset that delegates to this one

Reply:
https://github.com/pykeen/pykeen/pull/966/commits/2d38249cfa3c83261e0868b0827df2169ca3116d